### PR TITLE
Exception test modernization / clean up

### PR DESCRIPTION
**Pull request by offa**
A few test cases moved from try-catch to `assertThrows()`, commented out code removed (commented for quite some time) and minor clean up.

### Proposed changelog entries

* :question: Is this worth an entry?

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] ~~(If applicable) Jira issue is well described~~
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests --> Refactoring
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

-

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java:138]: That comment now makes no sense.

**Comment by daniel-beck** [comment]
> ❓ Is this worth an entry?

I don't think it is, no behavior changes in production code (just newly relying on implicit `toString` a lot). A few stack trace lines move but that's about it.

File: core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java
Comment (offa) [reply core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java:138]: Since the method is empty and it's base class provides an empty implementation already, is it safe to remove here?

Comment (daniel-beck) [reply core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java:138]: Alternatively revert the change; the commented out code looks to me to be more of a documentation nature at the moment, answering the question why there is nothing here.

Comment (offa) [reply core/src/main/java/jenkins/security/LastGrantedAuthoritiesProperty.java]: Done.

**Review by basil — APPROVED** [review]
Nice work chipping away at this test cleanup!

File: core/src/main/java/hudson/Proc.java
Comment (offa) [reply core/src/main/java/hudson/Proc.java]: Seems I was a bit too enthusiastic with removing … :smile:

Comment (daniel-beck) [reply core/src/main/java/hudson/Proc.java]: FWIW I was fine with this one, because the comment has everything of substance. Could go either way, so as it's back in, we can just keep it. 😃

**Review by daniel-beck — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
